src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -143,13 +150,48 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
         this.isSubmissionFormsDisabled = true;
       }
 
-      if (this.regKey) {
-        // public page using regKey, fetch CSRF token
-        this.authService.getAuthUser().subscribe(() => {}, () => {});
-      }
-
-      this.loadPersonName();
-      this.loadFeedbackSession();
+      const nextUrl: string = `${window.location.pathname}${window.location.search}`;
+      this.authService.getAuthUser(undefined, nextUrl).subscribe((auth: AuthInfo) => {
+        const isPreviewOrModeration: boolean = !!(auth.user && (this.moderatedPerson || this.previewAsPerson));
+        if (this.regKey && !isPreviewOrModeration) {
+          this.authService.getAuthRegkeyValidity(this.regKey, this.intent).subscribe((resp: RegkeyValidity) => {
+            if (resp.isValid) {
+              if (auth.user) {
+                // The logged in user matches the registration key; redirect to the logged in URL
+
+                const loggedInUrl: string = window.location.href.replace('/web/sessions', '/web/student/sessions')
+                    .replace(/key=[A-Z0-9]+(&?)/, '$1');
+                window.location.href = loggedInUrl;
```

Comment:
- I am wondering why `navigationService` is not used?
- In addition, we can be more specific: Instead of remove what is in the URL, we construct the URL more explicitly. In this case, I think we only need `courseid` and `fsname`